src/utils/error.hpp

```diff
@@ -31,15 +31,17 @@ namespace detail {
     // Lightweight std::source_location.
     struct source_location {
         const char* const file;
-        //const char* const function; // disabled for now due to static constexpr restrictions
+        // const char* const function; // disabled, because there are many different compiler definitions for it (see for example BOOST_CURRENT_FUNCTION)
         const int line;
         constexpr source_location(
-            //const char* _function /*= __builtin_FUNCTION()*/,
-            const char* _file     = __builtin_FILE(),
-            int _line             = __builtin_LINE()
-        ) : file(_file), /*function(_function),*/ line(_line) {}
+            const char* _file,
+            //const char* _function,
+            int _line
+        ) : file(_file)/*, function(_function)*/, line(_line) {}
     };
 
+    #define CPPTRACE_CURRENT_LOCATION ::cpptrace::detail::source_location(__FILE__, __LINE__)
```

Comment:
`PANIC` also uses `source_location`, though the macro isn't used in code that would be enabled under MSVC. For consistency, though, it'd be good to use this there too.